extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java

```diff
@@ -27,22 +30,22 @@ public class HeaderTest {
     @Test
     void testHeadersWithSubresource() {
         Client client = RestClientBuilder.newBuilder().baseUri(baseUri).build(Client.class);
-        assertThat(client.cookieSub("bar", "bar2").send("bar3", "bar4")).isEqualTo("bar:bar2:bar3:bar4");
+        assertThat(client.cookieSub("bar", "bar2").send("bar3", "bar4", "dummy")).isEqualTo("bar:bar2:bar3:bar4:dummy");
     }
 
     @Test
     void testNullHeaders() {
         Client client = RestClientBuilder.newBuilder().baseUri(baseUri).build(Client.class);
-        assertThat(client.cookieSub("bar", null).send(null, "bar4")).isEqualTo("bar:null:null:bar4");
+        assertThat(client.cookieSub("bar", null).send(null, "bar4", "dummy")).isEqualTo("bar:null:null:bar4:dummy");
     }
 
     @Path("/")
     @ApplicationScoped
     public static class Resource {
         @GET
         public String returnHeaders(@HeaderParam("foo") String header, @HeaderParam("foo2") String header2,
-                @HeaderParam("foo3") String header3, @HeaderParam("foo4") String header4) {
-            return header + ":" + header2 + ":" + header3 + ":" + header4;
+                @HeaderParam("foo3") String header3, @HeaderParam("foo4") String header4, @Context HttpHeaders headers) {
+            return header + ":" + header2 + ":" + header3 + ":" + header4 + ":" + headers.getHeaderString("X-My-Header");
```

Comment:
I am not entirely sure that this `getHeaderString` is not case-insensitive. In which case it's not testing anything ;)
In which case you'd be better off testing the keys of `getRequestHeaders`

Reply:
Yeah, you're probably right. That's why I checked with Wireshark, but lemme see if I can fix the test

Reply:
Should be good to go now